q2_metadata/normalization/_norm_rules.py

```diff
@@ -0,0 +1,322 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import yaml
+from glob import glob
+import pandas as pd
+from os.path import basename, isdir, splitext
+
+from q2_metadata.normalization._norm_messages import warningsCollection, errorsCollection
+
+
+class Rules(object):
+    """Collect the rules of one current variable rules.
+
+    Parameters
+    ----------
+    variable_rules_fp : str
+        Path to one variable's rules yaml file.
+
+    Attributes
+    ----------
+    variable : str
+        Name of the variable which rules are parsed and
+        that is the yaml file name without the .yml extension.
+
+    remap : dict
+        Apply the rules of remapping variables values, by replacing
+        all instances of a given value by another one value.
+                    'remap:
+                      str: str'
+    validation : dict
+        Apply the rules of validating variables values, by replacing by the blank
+        (or the missing) value every instance which in one or many other columns
+        have a null or missing value.
+                    'validation:
+                      force_to_blank_if:
+                        is null:
+                        - variable'
+    normalization : dict
+        Apply the rules of normalizing the variables values, by replacing
+        the values outside the range by a given replacement value.
+                    'normalization:
+                        gated_value:
+                            - Out of bounds
+                        maximum:
+                            - int
+                            - float
+                        minimum:
+                            - int
+                            - float'
+    missing : str
+        Apply the rules of checking the allowed missing values, by replacing
+        the not-expected values and numpy's nan (empty) by the passed
+        missing value term.
+                    'missing:
```

Comment:
are these the allowed missing values? can they be sync'd with qiita/ebi missing values?

Reply:
yes, the rules formatting checking could use a stricter controlled vocabulary - I'll check with Gail - but then, should it even be present in the yml files then? i.e. could the user choose which of the missing values to use?

Reply:
I think the currently allowed values (see new assets file: `q2_metadata/normalization/assets/rules_allowed.yml`) are thoses from qiita/ebi:
  - 'not applicable'
  - 'not collected'
  - 'not provided'
  - 'restricted access'